src/main/java/org/opensearch/security/user/UserService.java

```diff
@@ -314,4 +317,32 @@ public static void saveAndUpdateConfigs(
             throw ExceptionsHelper.convertToOpenSearchException(e);
         }
     }
+
+    /**
+     * Removes accounts that are not of the requested type from the SecurityDynamicConfiguration object passed.
+     *
+     * Accounts with the 'service' attribute set to true, are considered of type 'service'.
+     * Accounts with the 'service' attribute set to false or without the 'service' attribute, are considered of type 'internal'.
+     *
+     * @param configuration SecurityDynamicConfiguration object containing all accounts
+     * @param requestedAccountType The type of account to be kept. Should be "service" or "internal"
+     *
+     */
+    public void filterAccountsByType(SecurityDynamicConfiguration<?> configuration, String requestedAccountType) {
+        List<String> toBeRemoved = new ArrayList<>();
+
+        for (Map.Entry<String, ?> entry : configuration.getCEntries().entrySet()) {
+            final InternalUserV7 internalUserEntry = (InternalUserV7) entry.getValue();
```

Comment:
Since you are casting to InternalUserV7, there is a property `boolean service()`, can we use that instead?

Reply:
You can use the field. 

The `service` field is used when called with the internal users api.

For example:

```
    try {
            if (request.hasParam("service")) {
                ((ObjectNode) content).put("service", request.param("service"));
            }
            if (request.hasParam("enabled")) {
                ((ObjectNode) content).put("enabled", request.param("enabled"));
            }
            ((ObjectNode) content).put("name", username);
            internalUsersConfiguration = userService.createOrUpdateAccount((ObjectNode) content);
```